src/profiler.cpp

```diff
@@ -1710,6 +1724,14 @@ void Profiler::dumpOtlp(Writer& out, Arguments& args) {
         for (int j = 0; j < trace->num_frames; j++) {
             // To be written below in Profile.location_indices
             location_indices.push_back(functions.indexOf(fn.name(trace->frames[j])));
+            if (trace->frames[j].bci == BCI_TRACE_ID) {
+                const char* trace_id = (const char*)trace->frames[j].method_id;
+                printf("[dumpOtlp: ] trace_id: %s\n", trace_id ? trace_id : "NULL");
```

Comment:
Good, now we can go to the next step: https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/profiles/v1development/profiles.proto#L300

Please create `Link` messages using the protobuf writer (it's a relatively simple class, you find it in `protobuf.h`). You will also need to add more Protobuf indexes in `otlp.h`. Each `Link` should be part of the [`link_table`](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/profiles/v1development/profiles.proto#L111), and the corresponding index should go in [here](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/profiles/v1development/profiles.proto#L420).